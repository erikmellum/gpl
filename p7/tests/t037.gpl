string s1 = "one";
string s2 = "two";
string s3 = "three";
string s4 = "four";
string s5 = "five";
int i = 1;
double x = 1.1;

on akey
{
    s1 = s2;
    s2 = i;
    s3 = x;
    s4 = s1 + s2;
    s5 = i + x;
}

on skey
{
    print("s1 = " + s1);
    print("s2 = " + s2);
    print("s3 = " + s3);
    print("s4 = " + s4);
    print("s5 = " + s5);
}

