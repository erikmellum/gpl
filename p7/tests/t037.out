gpl.cpp::main()
  input file(tests/t037.gpl)
  random seed(42)
  read_keypresses_from_standard_input(true)
  dump_pixels(false)
  symbol_table(true)
  print_symbol_table(true)
  graphics(true)

gpl.cpp::main() Calling yyparse()


gpl.cpp::main() after call to yyparse().


Printing the symbol table from main()
int i 1
string s1 "one"
string s2 "two"
string s3 "three"
string s4 "four"
string s5 "five"
double x 1.1
gpl.cpp::main() Calling window->initialize().
gpl.cpp::main() Passing control to window->main_loop().
gpl[20]: s1 = two
gpl[21]: s2 = 1
gpl[22]: s3 = 1.1
gpl[23]: s4 = two1
gpl[24]: s5 = 2.1

Printing the symbol table from signal_handler()
int i 1
string s1 "two"
string s2 "1"
string s3 "1.1"
string s4 "two1"
string s5 "2.1"
double x 1.1
